### DISCUSSION TITLE
How is integration in axi-symmetric coordinates implemented?

### DISCUSSION THREAD

**Discussion by Minjiang-Zhu** [idaholab/moose#26751] [Q&A General] [answered]
Hi, I am writing some kernels generalized to cylindrical coordinate for axi-symmetric cases. Currently, the extra works I have done compared to simulation in cartesian coordinate include: 
1. Set "coord_type = RZ" in mesh module. 
2. Write integrand (both residual and jacobian) specialized for cylindrical coordinate. For example, there is an extra term u_r / r in velocity divergence.
However, as I understand, the integrand should be multiplied by r when applying integration. I want to confirm this is automatically implemented in MOOSE. May you check if I missed anything?

**Comment by GiudGiud** [comment, marked as answer]
Hello

1) is correct

2) Yes we automatically do that (and the 2pi factor too) when you specify coord_type of RZ
You are correct that you need to implement the extra term u_r / r

Guillaume
